src/Microsoft.AspNet.Authentication.OpenIdConnect/OpenIdConnectAuthenticationHandler.cs

```diff
@@ -197,6 +196,14 @@ public class OpenIdConnectAuthenticationHandler : AuthenticationHandler<OpenIdCo
 
             message.State = Options.StateDataFormat.Protect(properties);
 
+            if (!string.IsNullOrEmpty(Options.RedirectUri))
```

Comment:
Good point. I will update it to look into message.RedirectUri. If that is not set then fall back to Options.RedirectUri. I wanted to set it after the notificaiton is called to make sure it is send in the message and user has not removed it, like we do with state.